input/src/components/hooks.ts

```diff
@@ -201,3 +201,35 @@ export const useInputAssetProps = (
     value: localValue,
   };
 };
+
+/** Props for file type Input */
+export const useFileInputAssetProps = (
+  props: TransformedInput & { handleFile: (name: string) => void }
+) => {
+  /** Parses file content for upload into a string if file type Inpu */
+  const onFileUpload: React.ChangeEventHandler = (e): void => {
+    const fileList = (<HTMLInputElement>e.target).files;
+    const file = fileList ? fileList[0] : "";
+    const reader = new FileReader();
+
+    reader.addEventListener(
+      "load",
+      () => {
+        // this will set the file contents in the data model
+        props.set(reader.result as string);
+      },
+      false
+    );
+
+    if (file) {
+      reader.readAsText(file);
+      props.handleFile(file.name);
+    }
+  };
+
+  return {
+    type: "file",
+    onChange: onFileUpload,
+    accept: ".json, .txt",
```

Comment:
Should we allow the user to define those? We can default to `.json`, for example, but we could let them decide if they want to accept `.jsx` or `.tsx`.